url9e019a0ee9_elfrasco_design_principles_tgz-pJ8-ar_com_epidataconsulting_principles_ioc_before_SavePredictiveEngineProjectServiceJ8/src/ar/com/epidataconsulting/principles/model2/mock/EmployeeMockBuilder.java

```diff
@@ -0,0 +1,70 @@
+package ar.com.epidataconsulting.principles.model2.mock;
+
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+
+import ar.com.epidataconsulting.principles.model2.Analyst;
+import ar.com.epidataconsulting.principles.model2.Architect;
+import ar.com.epidataconsulting.principles.model2.Developer;
+import ar.com.epidataconsulting.principles.model2.Employee;
+import ar.com.epidataconsulting.principles.model2.ProjectManager;
+import ar.com.epidataconsulting.principles.model2.Tester;
+
+public class EmployeeMockBuilder {
+
+	public static Employee createFlorenciaCastagnino() {
+		Employee employee = new ProjectManager();
+		employee.setName("Florencia");
+		employee.setLastName("Castagnino");
+		employee.setDateOfBirth(new GregorianCalendar(1980, Calendar.JANUARY, 15).getTime());
+		employee.setFileNumber("123");
+		return employee;
+	}
+	
+	public static Employee createAdrianParedes() {
+		Employee employee = new Architect();
+		employee.setName("Adrian");
+		employee.setSecondName("Marcelo");
+		employee.setLastName("Paredes");
+		employee.setDateOfBirth(new GregorianCalendar(1982, Calendar.JUNE, 3).getTime());
+		employee.setFileNumber("456");
+		return employee;
+	}
+	
+	public static Employee createLeonardoPavanello() {
+		Employee employee = new Developer();
+		employee.setName("Leonardo");
+		employee.setLastName("Pavanello");
+		employee.setDateOfBirth(new GregorianCalendar(1984, Calendar.MAY, 28).getTime());
+		employee.setFileNumber("789");
+		return employee;		
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 35:40, 44:49.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.